sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -499,10 +504,17 @@ object GpuToTimestamp extends Arm {
       lhs: GpuColumnVector,
       sparkFormat: String,
       strfFormat: String,
-      dtype: DType): ColumnVector = {
+      dtype: DType,
+      failOnError: Boolean): ColumnVector = {
 
     // `tsVector` will be closed in replaceSpecialDates
     val tsVector = withResource(isTimestamp(lhs.getBase, sparkFormat, strfFormat)) { isTs =>
+      if(failOnError && !BoolUtils.isAllValidTrue(isTs)) {
+        // ANSI mode and has invalid value.
+        // CPU may throw `DateTimeParseException`, `DateTimeException` or `ParseException`
+        throw new IllegalArgumentException("Exception occurred when parsing timestamp in ANSI mode")
```

Comment:
Do we need a followup issue that we're not throwing a matching exception type here?

Reply:
I will add more tests when fixing this PR: https://github.com/NVIDIA/spark-rapids/issues/1556.
Do we need to distinguish these exceptions?  If so, let's do it in this PR https://github.com/NVIDIA/spark-rapids/issues/1556

Reply:
> Do we need to distinguish these exceptions? 

I don't think it is high priority to fix, and I could see adding some more tests while still leaving the exception types unaddressed.  So if #1556 does not address the exception types, we should file a low priority issue to track after that.

Reply:
Filed an issue to track: https://github.com/NVIDIA/spark-rapids/issues/5355